reference/host-crd.md

```diff
@@ -1,8 +1,167 @@
 
-The `Host` CRD defines how Ambassador will be visible to the  outside world. A minimal `Host` defines a hostname by which the Ambassador will be reachable, but a Host can also tell an Ambassador how to manage TLS, and which resources to examine for further configuration.
+A minimal Host resource, using Let’s Encrypt to handle TLS, would be:
+
+```yaml
+apiVersion: getambassador.io/v2
+kind: Host
+metadata:
+  name: minimal-host
+spec:
+  hostname: host.example.com
+  acmeProvider:
+    email: julian@example.com
+```
+
+This Host tells Ambassador to expect to be reached at `host.example.com`, and to manage TLS certificates using Let’s Encrypt, registering as `julian@example.com`. Since it doesn’t specify otherwise, requests using cleartext will be automatically redirected to use HTTPS, and Ambassador will not search for any specific further configuration resources related to this Host.
+
+## ACME and TLS Settings
+
+The acmeProvider element in a Host defines how Ambassador should handle TLS certificates:
+
+```yaml
+acmeProvider:
+  authority: url-to-provider
+  email: email-of-registrant
+  tlsSecret:
+    name: secret-name
+```
+
+* In general, `email-of-registrant` mandatory when using ACME: it should be a valid email address that will reach someone responsible for certificate management.
+* ACME stores certificates in Kubernetes secrets. The name of the secret can be set using the tlsSecret element; if not supplied, a name will be automatically generated from the hostname and email.
+* If the authority is not supplied, the Let’s Encrypt production environment is assumed.
+* **If the authority is the literal string “none”, TLS certificate management will be disabled.** You’ll need to manually create a TLSContext to use for your host in order to use HTTPS.
+
+## Secure and Insecure Requests
+
+A **secure** request arrives via HTTPS; an **insecure** request does not. By default, secure requests will be routed and insecure requests will be redirected (using an HTTP 301 response) to HTTPS. The behavior of insecure requests can be overridden using the requestPolicy element of a Host:
+
+```yaml
+requestPolicy:
+  insecure:
+    action: insecure-action
+    additionalPort: insecure-port
+```
+
+The `insecure-action` can be one of
+
+  `Redirect` (the default): redirect to HTTPS
+  `Route`: go ahead and route as normal; this will allow handling HTTP requests normally
+  `Reject`: reject the request with a 400 response
+
+If `additionalPort` is specified, Ambassador will listen on the specified `port` and treat any request arriving on that port as insecure.
+
+Some special cases to be aware of here:
+
+* The `X-Forwarded-Proto` header is honored when determining whether a request is secure or insecure. If you are running behind a load balancer, make sure that the `X-Forwarded-Proto` header is correctly set by your load balancer!
```

Comment:
I forgot to mention:

- **Case matters** in the actions: you must use e.g. `Reject`, not `reject`.